File: Competitive Programming 3/Section 1.3.3/10114 - Loansome Car Buyer/Java/Main.java
import java.util.Scanner;

public class Main {
    static class es{
        public int month;
        public double percentage;
        es(int month,double percentage){
            this.month=month;
            this.percentage=percentage;
        }
    };
    static es[] records= new es[108];
    static Scanner s = new Scanner(System.in);
    public static void main(String[] args){
        int months;
        while((months=s.nextInt())>=0){
            double downPayment = s.nextDouble(), loanAmount = s.nextDouble();
            int numberOfDepreRecords = s.nextInt();
            double carValue = downPayment + loanAmount;
            downPayment = loanAmount / months;
            int currentMonth = 0 ; 
            for(int i = 0 ; i < numberOfDepreRecords ; i ++){
                records[i] = new es(s.nextInt(),s.nextDouble());
            }
            records[numberOfDepreRecords] = new es(months+1,0);
            outer:
                for(int i = 1 ; i <= numberOfDepreRecords; i ++){
                    for(;currentMonth < records[i].month; currentMonth++){
                        carValue *= 1 - records[i-1].percentage;
                        if(loanAmount-currentMonth*downPayment<carValue){
                            System.out.printf("%d month%s\n",currentMonth,currentMonth!=1?"s":"");
                            break outer;   
                        }
                    }
                }
        }
    }
}

File: Competitive Programming 3/Section 1.3.3/10114 - Loansome Car Buyer/Python/main.py
class es:
    def __init__(self, month, percentage):
        self.month = month
        self.percentage = percentage
def main():   
    while(True):
        inTok = input().split()
        months = int(inTok[0])
        if(months<0):
            break
        downPayment,loanAmount,numberOfDepreRecords = float(inTok[1]), float(inTok[2]), int(inTok[3])
        carValue = downPayment + loanAmount
        downPayment = loanAmount / months
        currentMonth = 0 
        records = []
        for _ in range(0,numberOfDepreRecords):
            tk = input().split()
            records.append(es(int(tk[0]),float(tk[1])))
        records.append(es(months+1,0))
        for i in range(1,numberOfDepreRecords+1):
            done = False
            while(currentMonth<records[i].month):
                carValue *= 1 - records[i-1].percentage
                if loanAmount-currentMonth * downPayment < carValue :
                    print(str(currentMonth)+" month"+('s'if currentMonth!=1 else ''))
                    done = True
                    break
                currentMonth+=1
            if done : 
                break
main()